packages/salesforcedx-utils/test/jest/cli/cliCommandExecution.test.ts

```diff
@@ -0,0 +1,203 @@
+
+  beforeEach(() => {
+    testChildProcess = {
+      pid: 1234,
+      stdout: jest.fn(),
+      stderr: jest.fn()
+    };
+    testCancelationToken = {
+      isCancellationRequested: false
+    };
+    unsubscribeSpy = jest.fn();
+    subscribeSpy = jest.fn().mockReturnValue({
+      unsubscribe: unsubscribeSpy
+    });
+    fromEventSpy = jest.spyOn(Observable, 'fromEvent').mockReturnValue({
+      subscribe: subscribeSpy
+    } as any);
+    intervalSpy = jest.spyOn(Observable, 'interval').mockReturnValue({
+      subscribe: subscribeSpy
+    } as any);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('Should be able to create an instance.', () => {
+    const cliCommandExecution = new CliCommandExecution(
+      testCommand,
+      testChildProcess
+    );
+    expect(cliCommandExecution).toBeInstanceOf(CliCommandExecution);
+    expect(fromEventSpy).toHaveBeenCalledTimes(4);
+    expect(fromEventSpy.mock.calls[0]).toEqual([testChildProcess, 'exit']);
+    expect(fromEventSpy.mock.calls[1]).toEqual([testChildProcess, 'error']);
+    expect(fromEventSpy.mock.calls[2]).toEqual([
+      testChildProcess.stdout,
+      'data'
+    ]);
+    expect(fromEventSpy.mock.calls[3]).toEqual([
+      testChildProcess.stderr,
+      'data'
+    ]);
+
+    expect(subscribeSpy).toHaveBeenCalledTimes(2);
+  });
+
+  it('Should be able to create an instance with cancelation token.', () => {
+    const cliCommandExecution = new CliCommandExecution(
+      testCommand,
+      testChildProcess,
+      testCancelationToken
+    );
+    expect(cliCommandExecution).toBeInstanceOf(CliCommandExecution);
+    expect(subscribeSpy).toHaveBeenCalledTimes(3);
+    expect(intervalSpy).toHaveBeenCalledWith(CANCELATION_INTERVAL);
+  });
+
+  describe('Subscribe handlers.', () => {
+    it('Should call timer unsubscribe on exit.', () => {
+      const cliCommandExecution = new CliCommandExecution(
+        testCommand,
+        testChildProcess,
+        testCancelationToken
+      );
+      expect(cliCommandExecution).toBeDefined();
+      const exitSubscribeHandler = subscribeSpy.mock.calls[0][0];
+      exitSubscribeHandler();
+      expect(unsubscribeSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('Should call timer unsubscribe on error.', () => {
+      const cliCommandExecution = new CliCommandExecution(
+        testCommand,
+        testChildProcess,
+        testCancelationToken
+      );
+      expect(cliCommandExecution).toBeDefined();
+      const exitSubscribeHandler = subscribeSpy.mock.calls[1][0];
+      exitSubscribeHandler();
+      expect(unsubscribeSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Verify child process.', () => {
+    it('Should fail to create if child has no pid.', () => {
+      testChildProcess.pid = undefined;
+      expect(() => {
+        new CliCommandExecution(
+          testCommand,
+          testChildProcess,
+          testCancelationToken
+        );
+      }).toThrowError(NO_PID_ERROR);
```

Comment:
This is the syntax for verifying that a function call throws an error.  It's a little weird that you have to wrap the call in an arrow function (I almost always forget).

Reply:
👍 I'll follow on this PR with an update to tests.md